src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt

```diff
@@ -0,0 +1,198 @@
+                        source = source,
+                    )
+                },
+                author = doc.selectFirst("div.y6x11p i.fas.fa-user + span.dt")?.text()?.takeUnless {
+                    it.equals("updating", true)
+                },
+                state = when (doc.selectFirst("div.y6x11p i.fas.fa-rss + span.dt")?.text()?.lowercase()) {
+                    "ongoing", "đang tiến hành", "進行中" -> MangaState.ONGOING
+                    "completed", "hoàn thành", "完了" -> MangaState.FINISHED
+                    else -> null
+                },
+                chapters = doc.select("ul > li.chapter").mapChapters { i, element ->
+                    val href = element.selectFirst("a")?.attrAsRelativeUrl("href") ?: element.parseFailed("Chapter link not found")
+                    MangaChapter(
+                        id = generateUid(href),
+                        name = element.selectFirst("a")?.text().orEmpty(),
+                        number = i + 1f,
+                        url = href,
+                        scanlator = null,
+                        uploadDate = element.selectFirst("time[datetime]")?.attr("datetime")?.toLongOrNull()?.times(1000) ?: 0L,
+                        branch = null,
+                        source = source,
+                        volume = 0
+                    )
+                }
+            )
+        }
+
+        override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+            val fullUrl = chapter.url.toAbsoluteUrl(domain)
+            val doc = webClient.httpGet(fullUrl).parseHtml()
+            val script = doc.selectFirst("script:containsData(const CHAPTER_ID)")?.data()
+                ?: throw Exception("Failed to get chapter id")
+
+            val chapterId = script.substringAfter("const CHAPTER_ID = ").substringBefore(";")
+
+            val ajaxUrl = buildString {
+                append("https://")
+                append(domain)
+                append("/ajax/image/list/chap/")
+                append(chapterId)
+            }
+
+            val ajaxResponse = webClient.httpGet(ajaxUrl)
+            val responseJson = JSONObject(ajaxResponse.requireBody().string())
+
+            if (!responseJson.optBoolean("status", false)) {
+                throw Exception(responseJson.optString("msg"))
+            }
+
+            val pageListHtml = responseJson.getString("html")
+            val pageListDoc = org.jsoup.Jsoup.parse(pageListHtml)
+
+            return pageListDoc.select("div.separator[data-index]").map { page ->
+                val index = page.attr("data-index").toInt()
+                val url = page.selectFirst("a")?.attr("abs:href") ?: page.parseFailed("Image url not found")
+                MangaPage(
+                    id = generateUid(url),
+                    url = url,
+                    preview = null,
+                    source = source,
+                )
+            }.sortedBy { it.id.toInt() }
```

Comment:
`id` is a random number based on the `url` seed, sorting by it is not a good idea

Reply:
removed